src/UNL/Officefinder/Auth.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+class UNL_Officefinder_Auth
+{
+    const SESSION_MAP_CACHE_PREFIX = 'unl_cas_map_';
+
+    protected $cache;
+
+    public function __construct()
+    {
+        if (phpCAS::isInitialized()) {
+            return;
+        }
+
+        phpCAS::client(CAS_VERSION_2_0, 'login.unl.edu', 443, 'cas', false);
+        phpCAS::setPostAuthenticateCallback([$this, 'handleLoginTicket']);
+        phpCAS::setSingleSignoutCallback([$this, 'handleLogoutTicket']);
+        phpCAS::setCasServerCACert(GuzzleHttp\default_ca_bundle());
+    }
+
+    public function handleSingleLogOut()
+    {
+        phpCAS::handleLogoutRequests(false);
+    }
+
+    public function isAuthenticated()
+    {
+        return phpCAS::isAuthenticated();
+    }
+
+    public function getUsername()
+    {
+        return phpCAS::getUser();
+    }
+
+    public function gatewayAuthentication()
+    {
+        return phpCAS::checkAuthentication();
+    }
+
+    public function forceAuthentication()
+    {
+        return phpCAS::forceAuthentication();
+    }
+
+    public function logout($url = '')
+    {
+        return phpCAS::logout();
+    }
+
+    public function getLogoutUrl($url = '')
+    {
+        return phpCAS::getServerLogoutURL();
+    }
+
+    protected function getCache()
+    {
+        if (!$this->cache) {
+            $this->cache = UNL_Peoplefinder_Cache::factory();
+        }
+
+        return $this->cache;
+    }
+
+    public function handleLoginTicket($ticket)
+    {
+        $key = $this->getCacheKeyFromTicket($ticket);
+        $value = session_id();
+        $this->getCache()->set($key, $value);
+        return $this;
+    }
+
+    public function handleLogoutTicket($ticket)
+    {
+        $sessionId = $this->getSessionFromCache($ticket);
```

Comment:
I'm glad you ask. phpCAS uses the ticket from the CAS server as the new session id locally. This means that if someone learns your CAS ticket (which is trivial because it's in the querystring), they can hijack your session. By using our own generated session ids, we are doing a better job at keeping the login token secure and helping to ensure that the worst thing that can happen with the ticket is a forged logout request.